test/on.js

```diff
@@ -11,6 +11,7 @@ describe('on', function() {
   it('is a quaternary function', function() {
     eq(typeof S.on, 'function');
     eq(S.on.length, 4);
+    eq(S.on.toString(), 'on :: (b -> b -> c) -> (a -> b) -> a -> a -> c');
```

Comment:
It's exciting to be able to express the types of functions such as this one, @Bradcomp. :)